attest/secureboot.go

```diff
@@ -0,0 +1,166 @@
+package attest
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+
+	"github.com/google/certificate-transparency-go/x509"
+	"github.com/google/go-attestation/attest/internal"
+)
+
+// SecurebootState describes the secure boot status of a machine, as determined
+// by processing its event log.
+type SecurebootState struct {
+	Enabled bool
+
+	PlatformKeys      []x509.Certificate
+	PlatformKeyHashes [][]byte
+
+	ExchangeKeys      []x509.Certificate
+	ExchangeKeyHashes [][]byte
+
+	PermittedKeys   []x509.Certificate
+	PermittedHashes [][]byte
+
+	ForbiddenKeys   []x509.Certificate
+	ForbiddenHashes [][]byte
+
+	PreSeparatorAuthority  []x509.Certificate
+	PostSeparatorAuthority []x509.Certificate
+}
+
+// ExtractSecurebootState parses a series of events to determine the
+// configuration of secure boot on a device. An error is returned if
+// the state cannot be determined, or if the event log is structured
+// in such a way that it may have been tampered post-execution of
+// platform firmware.
+func ExtractSecurebootState(events []Event) (*SecurebootState, error) {
```

Comment:
We've got a lot of "Verify" "Parse" etc. synonyms. Instead of adding "Extract" can we just call this "ParseSecurebootState"?